File: kattis/inputoutput/different/different.py
import sys


class different:

    def __init__(self, debug: bool = False):
        self.debug = debug

    def dprint(self, *args, **kwargs):
        if self.debug:
            print(*args, **kwargs)

    def isValidNumber(self, inputNumber: int) -> bool:
        min = 0
        max = 1000000000000000  # 10^15
        return min <= inputNumber <= max

    def generateOutput(self):
        # Take input from user
        self.dprint("-------------------------------------")
        self.dprint("Input the file.")

        while True:
            inputLine = sys.stdin.readline()

            # EOF Check: stop loop if no more input is available
            if not inputLine:
                self.dprint("Reached end of file.")
                break

            # Find two numbers
            inputTwoNumbers = inputLine.strip().split()
            if len(inputTwoNumbers) != 2:
                self.dprint("Input count is not 2.")
                break

            firstNumber = int(inputTwoNumbers[0].strip())
            secondNumber = int(inputTwoNumbers[1].strip())
            self.dprint(
                f"First Number: {firstNumber} and Second Number: {secondNumber}"
            )

            # Check the validity of the numbers
            if not self.isValidNumber(firstNumber) or not self.isValidNumber(secondNumber):
            # if any(not self.isValidNumber(n) for n in [firstNumber, secondNumber]):
                self.dprint("Numbers are not valid.")
                break

            # Calculate the absolute value
            self.dprint("This is absolute value:")
            print(abs(firstNumber - secondNumber))

        self.dprint("-------------------------------------")


if __name__ == "__main__":
    different(False).generateOutput()

File: kattis/inputoutput/different/different.java
package kattis.inputoutput.different;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class different {

    private final boolean debug;

    public different(boolean debug) {
        this.debug = debug;
    }

    private void dprint(Object... args) {
        if (debug) {
            StringBuilder sb = new StringBuilder();
            for (Object arg : args) {
                sb.append(arg).append(" ");
            }
            System.out.println(sb.toString().trim());
        }
    }

    private boolean isValidNumber(long inputNumber) {
        long min = 0L;
        long max = 1000000000000000L; // 10^15
        return (inputNumber >= min && inputNumber <= max);
    }

    public void generateOutput() {
        try {
            BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
            String inputLine;

            // Take input from user
            dprint("-------------------------------------");
            dprint("Input the file.");

            // Loop until End of File (EOF)
            while ((inputLine = bufferedReader.readLine()) != null) {
                // Trim and tokenize the line string
                StringTokenizer stringTokenizer = new StringTokenizer(inputLine.trim());

                // Confirm the number of token
                if (stringTokenizer.countTokens() != 2) {
                    dprint("Input count is not 2.");
                    break;
                }

                // Find two numbers
                long firstNumber = Long.parseLong(stringTokenizer.nextToken().trim());
                long secondNumber = Long.parseLong(stringTokenizer.nextToken().trim());
                dprint("First Number: " + firstNumber + " and Second Number: " + secondNumber);

                // Check the validity of the numbers
                if (!isValidNumber(firstNumber) || !isValidNumber(secondNumber)) {
                    dprint("Numbers are not valid.");
                    break;
                }

                // Calculate the absolute value
                dprint("This is absolute value:");
                System.out.println(Math.abs(firstNumber - secondNumber));
            }

            dprint("Reached end of file.");
            dprint("-------------------------------------");
        } catch (Exception exception) {
            dprint(exception.getMessage());
        }
    }

    public static void main(String[] args) {
        new different(false).generateOutput();
    }
}